x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -630,6 +658,131 @@ export class TelemetryReceiver implements ITelemetryReceiver {
     return telemetryEvents;
   }
 
+  public async fetchTimelineEndpointAlerts(interval: number) {
+    if (this.esClient === undefined || this.esClient === null) {
+      throw Error('elasticsearch client is unavailable: cannot retrieve cluster infomation');
+    }
+
+    const query: SearchRequest = {
+      expand_wildcards: ['open' as const, 'hidden' as const],
+      index: `${this.alertsIndex}*`,
+      ignore_unavailable: true,
+      size: 100,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              {
+                bool: {
+                  should: [
+                    {
+                      match_phrase: {
+                        'event.module': 'endpoint',
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                bool: {
+                  should: [
+                    {
+                      match_phrase: {
+                        'kibana.alert.rule.parameters.immutable': 'true',
+                      },
+                    },
+                  ],
+                },
+              },
+              {
+                range: {
+                  '@timestamp': {
+                    gte: `now-${interval}h`,
+                    lte: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+      },
+    };
+
+    return this.esClient.search<EnhancedAlertEvent>(query);
+  }
+
+  public async buildProcessTree(
+    entityId: string,
+    resolverSchema: ResolverSchema,
+    startOfDay: string,
+    endOfDay: string
+  ): Promise<ResolverNode[]> {
+    if (this.processTreeFetcher === undefined || this.processTreeFetcher === null) {
+      throw Error(
+        'resolver tree builder is unavailable: cannot build encoded endpoint event graph'
+      );
+    }
+
+    const request: TreeOptions = {
+      ancestors: 200,
+      descendants: 500,
+      timeRange: {
+        from: startOfDay,
+        to: endOfDay,
+      },
+      schema: resolverSchema,
+      nodes: [entityId],
+      indexPatterns: [`${this.alertsIndex}*`, 'logs-*'],
+      descendantLevels: 20,
+    };
+
+    return this.processTreeFetcher.tree(request, true);
+  }
+
+  public async fetchTimelineEvents(nodeIds: string[]) {
+    if (this.esClient === undefined || this.esClient === null) {
+      throw Error('elasticsearch client is unavailable: cannot retrieve timeline endpoint events');
+    }
+
+    const query: SearchRequest = {
+      expand_wildcards: ['open' as const, 'hidden' as const],
+      index: [`${this.alertsIndex}*`, 'logs-*'],
+      ignore_unavailable: true,
+      size: 100,
```

Comment:
Does this mean we get 100 graph nodes per 3h?

Reply:
No, Max 100 EP alerts (regardless of node size in tree) every 3 hours for 8.3, sent 1 at a time (To max the 10MB budget).
We will budget when we see how this performs during our post-FF testing.